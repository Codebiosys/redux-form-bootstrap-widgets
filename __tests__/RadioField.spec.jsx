/* eslint-disable import/first */

import React from 'react';
import { mount } from 'enzyme';
import { omit } from 'lodash';
import { RadioField } from 'index';

const fieldProps = {
  input: {
    name: 'fieldName',
    onChange: jest.fn(),
    onBlur: jest.fn(),
    onFocus: jest.fn(),
  },
  meta: {
    form: 'formName',
    pristine: false,
  },
  label: 'Radio Test',
  helpText: 'The help Text',
  options: [
    { label: 'Option 1', value: 'One' },
    { label: 'Option 2', value: 'Two' },
    { label: 'Option 3', value: 'Three' },
  ],
};

const selectedFieldProps = { ...fieldProps };
selectedFieldProps.input = { ...fieldProps.input };
selectedFieldProps.input.value = 'One';

const customOptionFieldProps = { ...fieldProps };
customOptionFieldProps.labelKey = 'foo';
customOptionFieldProps.valueKey = 'bar';
customOptionFieldProps.options = [
  { foo: 'Option 1', bar: 'One' },
  { foo: 'Option 2', bar: 'Two' },
  { foo: 'Option 3', bar: 'Three' }];

describe('The Radio Field', () => {
  let inputWrapper;

  beforeEach(() => {
    inputWrapper = mount(<RadioField {...fieldProps} />);
  });

  it('renders', () => {
    expect(inputWrapper).toMatchSnapshot();
  });

  it('has a label when there is a label', () => {
    expect(inputWrapper.find('ControlLabel').text()).toEqual(fieldProps.label);
  });

  it('renders with custom options', () => {
    inputWrapper = mount(<RadioField {...customOptionFieldProps} />);
    expect(inputWrapper).toMatchSnapshot();
  });
  it('triggers the focus when focused', () => {
    inputWrapper.find('input[value="One"]').simulate('focus');
    expect(fieldProps.input.onFocus).toHaveBeenCalledTimes(1);
  });

  it('triggers the blur when blurred', () => {
    inputWrapper.find('input[value="One"]').simulate('blur');
    expect(fieldProps.input.onBlur).toHaveBeenCalledTimes(1);
  });
  it('calls onChange when the radio is selected', () => {
    inputWrapper.find('input[value="One"]').simulate('click');
    expect(fieldProps.input.onChange).toHaveBeenCalledTimes(1);
    expect(fieldProps.input.onBlur).toHaveBeenCalledTimes(1);
    expect(fieldProps.input.onChange).toHaveBeenCalledWith('One');
  });

  it('clears value  when the radio is deselected', () => {
    inputWrapper = mount(<RadioField {...selectedFieldProps} />);
    inputWrapper.find('input[value="One"]').simulate('click');
    expect(fieldProps.input.onChange).toHaveBeenCalledTimes(1);
    expect(fieldProps.input.onChange).toHaveBeenCalledWith(null);
  });

  it('calls custom validator when the toogle is toggled', () => {
    const customValidator = jest.fn(() => ({ validationState: null, errorMessage: null }));
    const validatorProps = { ...fieldProps, customValidation: customValidator };
    const inputWrapperValidated = mount(<RadioField {...validatorProps} />);

    inputWrapperValidated.find('input[value="One"]').simulate('click');
    expect(customValidator).toHaveBeenCalledTimes(1);
  });
});
