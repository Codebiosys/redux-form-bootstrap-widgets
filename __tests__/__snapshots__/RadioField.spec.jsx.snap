// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Radio Field renders 1`] = `
<RadioField
  helpText="The help Text"
  input={
    Object {
      "name": "fieldName",
      "onBlur": [Function],
      "onChange": [Function],
      "onFocus": [Function],
    }
  }
  label="Radio Test"
  meta={
    Object {
      "form": "formName",
      "pristine": false,
    }
  }
  options={
    Array [
      Object {
        "label": "Option 1",
        "value": "One",
      },
      Object {
        "label": "Option 2",
        "value": "Two",
      },
      Object {
        "label": "Option 3",
        "value": "Three",
      },
    ]
  }
>
  <FormGroup
    bsClass="form-group"
    controlId="fieldName"
  >
    <div
      className="form-group"
    >
      <Component
        label="Radio Test"
      >
        <ControlLabel
          bsClass="control-label"
          srOnly={false}
        >
          <label
            className="control-label"
            htmlFor="fieldName"
          >
            Radio Test
          </label>
        </ControlLabel>
      </Component>
      <InputGroup
        bsClass="input-group"
      >
        <span
          className="input-group"
        >
          <div>
            <Radio
              bsClass="radio"
              checked={false}
              disabled={false}
              inline={false}
              name="fieldName"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              title=""
              value="One"
            >
              <div
                className="radio"
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={false}
                    name="fieldName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    type="radio"
                    value="One"
                  />
                  Option 1
                </label>
              </div>
            </Radio>
          </div>
          <div>
            <Radio
              bsClass="radio"
              checked={false}
              disabled={false}
              inline={false}
              name="fieldName"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              title=""
              value="Two"
            >
              <div
                className="radio"
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={false}
                    name="fieldName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    type="radio"
                    value="Two"
                  />
                  Option 2
                </label>
              </div>
            </Radio>
          </div>
          <div>
            <Radio
              bsClass="radio"
              checked={false}
              disabled={false}
              inline={false}
              name="fieldName"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              title=""
              value="Three"
            >
              <div
                className="radio"
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={false}
                    name="fieldName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    type="radio"
                    value="Three"
                  />
                  Option 3
                </label>
              </div>
            </Radio>
          </div>
        </span>
      </InputGroup>
      <HelpBlock
        bsClass="help-block"
      >
        <span
          className="help-block"
        >
          The help Text
        </span>
      </HelpBlock>
    </div>
  </FormGroup>
</RadioField>
`;

exports[`The Radio Field renders with custom options 1`] = `
<RadioField
  helpText="The help Text"
  input={
    Object {
      "name": "fieldName",
      "onBlur": [Function],
      "onChange": [Function],
      "onFocus": [Function],
    }
  }
  label="Radio Test"
  labelKey="foo"
  meta={
    Object {
      "form": "formName",
      "pristine": false,
    }
  }
  options={
    Array [
      Object {
        "bar": "One",
        "foo": "Option 1",
      },
      Object {
        "bar": "Two",
        "foo": "Option 2",
      },
      Object {
        "bar": "Three",
        "foo": "Option 3",
      },
    ]
  }
  valueKey="bar"
>
  <FormGroup
    bsClass="form-group"
    controlId="fieldName"
  >
    <div
      className="form-group"
    >
      <Component
        label="Radio Test"
      >
        <ControlLabel
          bsClass="control-label"
          srOnly={false}
        >
          <label
            className="control-label"
            htmlFor="fieldName"
          >
            Radio Test
          </label>
        </ControlLabel>
      </Component>
      <InputGroup
        bsClass="input-group"
      >
        <span
          className="input-group"
        >
          <div>
            <Radio
              bsClass="radio"
              checked={false}
              disabled={false}
              inline={false}
              name="fieldName"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              title=""
              value="One"
            >
              <div
                className="radio"
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={false}
                    name="fieldName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    type="radio"
                    value="One"
                  />
                  Option 1
                </label>
              </div>
            </Radio>
          </div>
          <div>
            <Radio
              bsClass="radio"
              checked={false}
              disabled={false}
              inline={false}
              name="fieldName"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              title=""
              value="Two"
            >
              <div
                className="radio"
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={false}
                    name="fieldName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    type="radio"
                    value="Two"
                  />
                  Option 2
                </label>
              </div>
            </Radio>
          </div>
          <div>
            <Radio
              bsClass="radio"
              checked={false}
              disabled={false}
              inline={false}
              name="fieldName"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              title=""
              value="Three"
            >
              <div
                className="radio"
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={false}
                    name="fieldName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    type="radio"
                    value="Three"
                  />
                  Option 3
                </label>
              </div>
            </Radio>
          </div>
        </span>
      </InputGroup>
      <HelpBlock
        bsClass="help-block"
      >
        <span
          className="help-block"
        >
          The help Text
        </span>
      </HelpBlock>
    </div>
  </FormGroup>
</RadioField>
`;
